File: scripts/apply-images.js
// Patches data.js imageUrl fields with the verified Wikipedia/Commons URLs
const fs = require('fs');

const urls = JSON.parse(fs.readFileSync('/tmp/paint-image-urls.json', 'utf8'));
let dataJs = fs.readFileSync('/Users/camfaux/Vibe_Projects/Paint Chips/data.js', 'utf8');

let patched = 0;
let failed = 0;

for (const [id, url] of Object.entries(urls)) {
  if (!url) { failed++; continue; }

  // Match the painting entry block by its id field, then replace imageUrl
  // Pattern: find `id: NUMBER,` ... `imageUrl: 'ANYTHING',`
  // We use a regex that finds the imageUrl line within the painting's object block
  // Strategy: replace imageUrl for each painting by finding the block starting with `  { id: N,`
  const idNum = parseInt(id);

  // Match the specific painting's imageUrl line — look for id followed by imageUrl within ~30 lines
  const blockRegex = new RegExp(
    `(\\{[^}]*?id:\\s*${idNum},[^}]*?imageUrl:\\s*')[^']*('(?:[^}]*?\\}))`,
    's'
  );

  const escaped = url.replace(/\\/g, '\\\\').replace(/'/g, "\\'");
  const newData = dataJs.replace(blockRegex, `$1${escaped}$2`);

  if (newData === dataJs) {
    console.log(`✗ ${id}: regex did not match`);
    failed++;
  } else {
    dataJs = newData;
    patched++;
  }
}

fs.writeFileSync('/Users/camfaux/Vibe_Projects/Paint Chips/data.js', dataJs);
console.log(`\nPatched: ${patched}, Failed: ${failed}`);

File: scripts/apply-images.py
import json, re

with open('/tmp/paint-image-urls.json') as f:
    urls = json.load(f)

with open('/Users/camfaux/Vibe_Projects/Paint Chips/data.js') as f:
    lines = f.readlines()

patched = 0
current_id = None

for i, line in enumerate(lines):
    # Track current painting id
    id_match = re.search(r'id:\s*(\d+),', line)
    if id_match:
        current_id = str(id_match.group(1))

    # Replace imageUrl line for current painting
    if current_id and 'imageUrl:' in line and current_id in urls and urls[current_id]:
        new_url = urls[current_id]
        # Preserve indentation and format
        indent = re.match(r'^(\s*)', line).group(1)
        lines[i] = f'{indent}imageUrl: "{new_url}",\n'
        patched += 1

with open('/Users/camfaux/Vibe_Projects/Paint Chips/data.js', 'w') as f:
    f.writelines(lines)

print(f'Patched {patched}/100 imageUrl entries')
